pkg/accounting/accounting.go

```diff
@@ -168,35 +155,29 @@ func (a *Accounting) Reserve(ctx context.Context, peer swarm.Address, price uint
 	}
 
 	// uint64 conversion of surplusbalance is safe because surplusbalance is always positive
-	if additionalDebt < 0 {
+	if additionalDebt.Cmp(big.NewInt(0)) < 0 {
 		return ErrInvalidValue
 	}
 
-	increasedExpectedDebt, err := addI64pU64(expectedDebt, uint64(additionalDebt))
-	if err != nil {
-		return err
-	}
+	increasedExpectedDebt := new(big.Int).Add(expectedDebt, additionalDebt)
 
 	// If our expected debt is less than earlyPayment away from our payment threshold
 	// and we are actually in debt, trigger settlement.
 	// we pay early to avoid needlessly blocking request later when concurrent requests occur and we are already close to the payment threshold.
-	if increasedExpectedDebt >= int64(threshold) && currentBalance < 0 {
+	if increasedExpectedDebt.Cmp(threshold) >= 0 && currentBalance.Cmp(big.NewInt(0)) < 0 {
```

Comment:
i think that you could extract a package level `big.NewInt(0)` and reuse it in all of the comparators that check whether a value is smaller/bigger than zero

Reply:
Will look into this in a follow up. This variable will be accessed from many threads but since all of them only read its probably fine.